### PR TITLE
Conditionally add scrollView on iOS instead of pager

### PR DESCRIPTION
**Pull request by osdnk**
### Motivation

Might deliver a bit better UX than normal pager. This was considered as a regression compared to v.1

### Test plan

Run the example and then `CustomIndicatorExample`

### CODE REVIEW

File: src/IOSPager.tsx [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    if (this.props.layout.width) {
+      this.scrollTo(
+        this.props.navigationState.index * this.props.layout.width,
+        false
+      );
+    }
+  }
+
+  componentDidUpdate(prevProps: Props<T>) {
+    const amount = this.props.navigationState.index * this.props.layout.width;
```
Comment (WoLewicki) [inline src/IOSPager.tsx]:
```suggestion
    const offset = this.props.navigationState.index * this.props.layout.width;
```

File: README.md [hunk 2]
```diff
@@ -285,6 +285,12 @@ If this is not specified, the default tab bar is rendered. You pass this props t
 
 Callback which returns a custom React Element to use as pager:
 
+##### `enabledIOSScrollViewPager`
```
Comment (satya164) [inline README.md]: I think no need for a separate prop. Isn't `renderPager` enough for it?

File: README.md [hunk 3]
```diff
@@ -285,6 +285,12 @@ If this is not specified, the default tab bar is rendered. You pass this props t
...
+When `true` renders `ScrollView` instead of default pager. It might deliver slightly better experience. However, `gestureHandlerProps` have no impact then. Defaults to `false`. 
+
+##### `overscroll`
```
Comment (satya164) [inline README.md]: Same as above. Props could be passed to the pager directly

File: src/IOSPager.tsx [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+  initialOffset: { x: number; y: number };
+};
+
+declare module 'react-native' {
+  export interface ScrollViewProps {
+    scrollToOverflowEnabled?: boolean;
+  }
+}
+
+export default class IOSPager<T extends Route> extends React.Component<
```
Comment (satya164) [inline src/IOSPager.tsx]: Can we call it `ScrollPager` instead?

File: src/IOSPager.tsx [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+import { Route, Listener } from './types';
+import { Props } from './Pager';
+
+const { event, divide } = Animated;
+
+type State = {
+  initialOffset: { x: number; y: number };
+};
+
+declare module 'react-native' {
```
Comment (satya164) [inline src/IOSPager.tsx]: Not sure about this. Shouldn't this be fixed in RN?

File: src/IOSPager.tsx [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+  }
+
+  private initialOffset = {
+    x: this.props.navigationState.index * this.props.layout.width,
+    y: 0,
+  };
+
+  private scrollViewRef: React.RefObject<
+    Animated.ScrollView
+  > = React.createRef();
```
Comment (satya164) [inline src/IOSPager.tsx]: `private scrollViewRef = React.createRef<Animated.ScrollView>();` might be enough

File: src/DefaultPager.tsx [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
```
Comment (satya164) [inline src/DefaultPager.tsx]:
```suggestion
import * as React from 'react';
```

File: src/IOSPager.tsx [hunk 8]
```diff
@@ -0,0 +1,181 @@
+import * as React from 'react';
+import { StyleSheet, Keyboard } from 'react-native';
+import Animated from 'react-native-reanimated';
+
+import { Route, Listener } from './types';
```
Comment (satya164) [inline src/IOSPager.tsx]: nit: type imports at end

File: src/IOSPager.tsx [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+      children,
+      layout,
+      onSwipeStart,
+      onSwipeEnd,
+      overscroll,
+      navigationState,
+    } = this.props;
+
+    return children({
+      position: divide(this.position, layout.width),
```
Comment (satya164) [inline src/IOSPager.tsx]: Maybe we need to move this node to a property to not create new node every render to avoid the reanimated bridge issue?

**Pull request merged**